src/batch/src/task/task_execution.rs

```diff
@@ -544,9 +544,19 @@ impl<C: BatchTaskContext> BatchTaskExecution<C> {
                               break;
                         },
                         None => {
-                             debug!("Batch task {:?} finished successfully.", self.task_id);
-                             state = TaskStatus::Finished;
-                             break;
+                            if self.context.get_stop_flag_ref().check_stop() {
+                                // Task stop by reading stop flag. It can be either cancelled or aborted.
+                                if let Some(error_reason) = self.context.get_stop_flag_ref().get_stop_reason() {
+                                    state = TaskStatus::Aborted;
+                                    error = Some(Aborted(error_reason));
+                                } else {
+                                    state = TaskStatus::Cancelled;
+                                }
```

Comment:
We will add cancel_flag support for executors one by one, so I think enable this check after all executors have supported it, otherwise I'm not sure whether we may introduce any inconsistent behavior.